### DISCUSSION TITLE
Need help with appropriate usage of DataFilterExtension filter_categories

### DISCUSSION THREAD

**Discussion by ATL2001** [developmentseed/lonboard#711] [Q&A]
I'm not sure what I'm doing wrong, but I can't get the points on my map to filter using filter_categories.

this is the code I'm using to make my data and layer with the data filter extension add it to my Lonboard map
```python
import geopandas as gpd
import ipywidgets 
from shapely.geometry import Point

import lonboard
from lonboard.basemap import CartoBasemap
from lonboard.layer_extension import DataFilterExtension

d = {
    'col_to_filter': [0.0, 1.0, 2.0, 3.0, 4.0, 5.0], 
    'geometry': [
        Point(0, 0), 
        Point(1, 1), 
        Point(2, 2), 
        Point(3, 3), 
        Point(4, 4), 
        Point(5, 5)
    ]
}
gdf = gpd.GeoDataFrame(d, crs="EPSG:4326")

dfe = DataFilterExtension(filter_size=1, category_size=1)
point_layer = lonboard.ScatterplotLayer.from_geopandas(
    gdf,
    get_fill_color=(0,255,0),
    radius_min_pixels=10,
    extensions=[dfe],
    get_filter_category=gdf["col_to_filter"]
)

m = lonboard.Map(layers=[point_layer], basemap_style=CartoBasemap.DarkMatter)
ipywidgets.VBox([m])
```

and then after I create the map, I was thinking that this should result in only the point at (2,2) being displayed, but when I run it, they're all still there :( 
```python
point_layer.filter_categories = [2]
```

can anyone help?
![image](https://github.com/user-attachments/assets/4391137d-0fd9-435f-89b1-f5dc9f7606a3)

**Comment by kylebarron** [comment]
I think the underlying deck.gl filter categories is more meant for non-float data. It might be an issue with float comparisons here, especially if `2` is uploaded once to the GPU as a float32 and maybe another time as an integer, or something like that.

If you want to choose a specific category that's based on a float value, perhaps first consider a min range and max range that are within an epsilon of that value. 

Maybe try `filter_categories` with a string value? If we get a working example we should add it to the docs, because I know it's a little finnicky

**Comment by ATL2001** [comment]
yeah, I'll certainly make up an example notebook if I get it working! I tried using strings after looking at the deck.gl docs but when I pass that column from my dataframe to the get_filter_category parameter I end up with a TraitError.

```python-exception
TraitError: The 'get_filter_category' trait of a ScatterplotLayer instance expected numeric dtype, not the ndarray array(['0', '1', '2', '3', '4', '5'], dtype=object)
```

I may tinker around this weekend with the validate funciton in traits.py to not check for numeric dtypes and see if I can get it to work with strings.

**Reply by kylebarron** [reply]
That sounds awesome, thanks 🙏
